src/search.cpp

```diff
@@ -928,18 +928,18 @@ namespace {
       if (    depth >= 8 * ONE_PLY
           &&  move == ttMove
           && !rootNode
-          && !excludedMove // Recursive singular search is not allowed
+          && !excludedMove // needed to avoid recursive singular search
```

Comment:
@protonspring  
This still suggest that we don't want recursive singular search, which isn't the case.
If my proposal 'needed to avoid a vicious circle' is not accepted, then as alternative we could write
'needed to avoid recursive singular search on the same node' or just remove the whole comment 
(the code here is self-explaining enough IMO, the comment just confuses the ideas).